lib/localize.js

```diff
@@ -164,8 +191,15 @@ export const getLocalizeClass = (superclass = class {}) => class LocalizeClass e
 		}
 		if (Object.prototype.hasOwnProperty.call(this, 'getLocalizeResources') || Object.prototype.hasOwnProperty.call(this, 'resources')) {
 			const possibleLanguages = this._generatePossibleLanguages(config);
-			const resourcesPromise = this.getLocalizeResources(possibleLanguages, config);
-			resourcesLoadedPromises.push(resourcesPromise);
+			if (config?.importFunc) {
+				const resourcesPromise = this.getLocalizeResources(possibleLanguages, config);
+				resourcesLoadedPromises.push(resourcesPromise);
+			}
+			if (config?.loadCommon) {
```

Comment:
This is now only loading the common resources if the consumer explicitly asks for it.